src/main/java/io/jenkins/docker/DockerTransientNode.java

```diff
@@ -7,44 +7,51 @@
 import com.nirima.jenkins.plugins.docker.strategy.DockerOnceRetentionStrategy;
 import hudson.model.Computer;
 import hudson.model.Descriptor;
-import hudson.model.Queue;
 import hudson.model.Slave;
 import hudson.model.TaskListener;
 import hudson.slaves.Cloud;
 import hudson.slaves.ComputerLauncher;
 import io.jenkins.docker.client.DockerAPI;
+import io.jenkins.docker.connector.DockerContainerComputerLauncher;
 import jenkins.model.Jenkins;
+import org.apache.commons.lang.StringUtils;
 
-import javax.annotation.CheckForNull;
 import javax.annotation.Nonnull;
 import java.io.IOException;
-import java.io.Serializable;
 
 /**
  * A {@link Slave} node designed to be used only once for a build.
  * 
  * @author <a href="mailto:nicolas.deloof@gmail.com">Nicolas De Loof</a>
  */
 public class DockerTransientNode extends Slave {
-    
-    private final String containerId;
-    
+
```

Comment:
please do not remove this useful information. Just make this non-final and assign as container has been created

Reply:
Restored.